cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,225 @@
+"""Contains cog classes for annual handover and role reset functionality."""
+
+
+from collections.abc import Sequence
+
+__all__: Sequence[str] = ("HandoverCommandCog", "AnnualResetRolesCommandCog")
+
+import logging
+from logging import Logger
+from typing import Final
+
+import discord
+
+from db.core.models import GroupMadeMember
+from utils import AllChannelTypes, CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class HandoverCommandCog(TeXBotBaseCog):
+    """Cog class that defines the handover command."""
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="handover",
+        description="Initiates the annual Discord handover procedure for new committee",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def handover(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "handover" command.
+
+        The "handover" command runs the relevant handover methods
+        which will perform the following actions:
+        - Give the "Committee" role to any users that have the "Committee-Elect" role
+        - Remove the "Committee-Elect" role from any user that has it
+        - Remove the permissions for the "Committee-Elect" role
+          from all channels except "#handover"
+
+        To do this, TeX-Bot will need to hold a role above that of the committee role.
```

Comment:
Reference to Committee role requires speech marks & capitalisation: `...the "Committee" role.`